toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -791,3 +808,153 @@ func getPartitionNum(partitionLoopDevice string) (int, error) {
+	mount, err := safemount.NewMount(devicePath, mountPoint, "btrfs", unix.MS_RDONLY, mountOptions, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to mount BTRFS subvolume %s:\n%w", subvolume, err)
+	}
+	return mount, nil
+}
+
+// processBtrfsSubvolumes is a helper function that handles the common pattern of mounting BTRFS,
+// listing subvolumes, and iterating through them with a callback function
+func processBtrfsSubvolumes(diskPartition *diskutils.PartitionInfo, tmpDir string,
+	callback func(subvolume string, subvolMount *safemount.Mount) (bool, error),
+) (bool, error) {
+	// First mount the BTRFS filesystem to discover subvolumes
+	partitionMount, err := safemount.NewMount(diskPartition.Path, tmpDir, diskPartition.FileSystemType,
+		unix.MS_RDONLY, "", true)
+	if err != nil {
+		return false, fmt.Errorf("failed to mount partition (%s):\n%w", diskPartition.Path, err)
+	}
+	defer partitionMount.Close()
+
+	// List actual subvolumes using btrfs tools
+	subvolumes, err := listBtrfsSubvolumes(tmpDir)
+	if err != nil {
+		logger.Log.Debugf("Failed to list BTRFS subvolumes for partition %s:\n%v", diskPartition.Path, err)
+		// Fallback to checking default location with callback
+		return callback("", partitionMount)
+	}
+
+	// Close the current mount and mount each subvolume individually
+	err = partitionMount.CleanClose()
+	if err != nil {
+		return false, fmt.Errorf("failed to close partition mount (%s):\n%w", diskPartition.Path, err)
+	}
+
+	// Check each subvolume
+	for _, subvolume := range subvolumes {
+		subvolMount, err := mountBtrfsSubvolume(diskPartition.Path, subvolume, tmpDir)
+		if err != nil {
+			logger.Log.Debugf("Failed to mount BTRFS subvolume %s:\n%v", subvolume, err)
+			continue
+		}
+
+		found, err := callback(subvolume, subvolMount)
+		subvolMount.Close() // Always close the subvolume mount
+
+		if err != nil {
+			logger.Log.Debugf("Error processing BTRFS subvolume %s:\n%v", subvolume, err)
+			continue
+		}
+
+		if found {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// findRootfsInBtrfsPartition searches for fstab in BTRFS subvolumes to determine if this is the rootfs partition
+func findRootfsInBtrfsPartition(diskPartition *diskutils.PartitionInfo, tmpDir string) (bool, error) {
+	return processBtrfsSubvolumes(diskPartition, tmpDir, func(subvolume string, mount *safemount.Mount) (bool, error) {
+		// Check for fstab in this subvolume (or default mount if subvolume is empty)
+		fstabPath := filepath.Join(tmpDir, "etc/fstab")
+		exists, err := file.PathExists(fstabPath)
+		if err != nil {
+			return false, err
+		}
+
+		if exists && subvolume != "" {
+			logger.Log.Debugf("Found fstab in BTRFS subvolume: %s", subvolume)
+		}
+
+		return exists, nil
+	})
+}
+
+// readFstabFromBtrfsPartition reads fstab entries from a BTRFS partition by searching through subvolumes
+func readFstabFromBtrfsPartition(rootfsPartition *diskutils.PartitionInfo, tmpDir string) ([]diskutils.FstabEntry,
```

Comment:
Return the subvolume from `findRootfsPartition` so that you don't have to find the subvolume again.